src/query/query_eval.rs

```diff
@@ -7,7 +7,7 @@ use serde_json::value::Value as JsonValue;
 pub fn eval_query<'a>(index: &'a Index, query: QueryNode) -> Result<Vec<&'a Packet>, QueryError> {
     match query {
         QueryNode::Latest(inner) => eval_latest(index, inner),
-        QueryNode::Lookup(field, value) => eval_lookup(index, field, value),
+        QueryNode::Test(test, field, value) => eval_test(index, test, value, field),
```

Comment:
I've updated some of the naming here to be a bit more in line with the R impl and also I think makes it a bit clearer! Best perhaps to look at the query_types to understand this but essentially now there are `QueryNode`'s which are a `Test` which is one of =, !=, <, <=, >, >= and either side of this test can be a `Literal` (i.e. the was submitted on the command line) or a `Lookup` (name, id, parameter:x)

Reply:
Not sure what you mean by either side - that the Lookup and Literal can be on either side of the operator, but there should always be one of each? Or that you could have two Lookups e.g. to get results where param1 > param2? The signature of `eval_test` and the pest definition suggests that you've always got a lookup LHS and a literal RHS?

Reply:
Sorry bit confusing language above. By either side I just meant in e.g. `a < b` that a and b are literally either side of the test `<`. But in the sentence above what I really wanted to say was that `Lookup` now has a different meaning to before. Before the `QueryNode::Lookup` was really a test that a field matched some value. So it would lookup the a value from the field name, then test it for equality. Now reworded this in the code so that it is a QueryNode::Test where one of the args is a new `Lookup` type and the other is a `Literal`.

You're right that at the moment the Lookup has to be on the LHS and the Literal on the RHS of the test. There is a (ticket)[https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-4531/Make-query-lookups-work-in-any-order] to support the reverse order